File: code/12-JavaCodeLoop/JavaTDDLoop/src/main/java/com/example/domain/NumberGuesser.java
package com.example.domain;

import com.example.valueobjects.Interval;

/**
 * Produces the next guess for a {@code GuessGame}.
 *
 * <p>Guesses the midpoint of the interval (binary search).</p>
 */
public class NumberGuesser {

    public static NumberGuesser create() {
        return new NumberGuesser();
    }

    public int nextGuess(Interval interval) {
        return interval.lower() + (interval.upper() - interval.lower()) / 2;
    }
}

File: code/13-Java2Python/python/domain/number_guesser.py
from valueobjects.interval import Interval


class NumberGuesser:
    """Produces the next guess for a GuessGame.

    Guesses the midpoint of the interval (binary search).
    """

    @staticmethod
    def create() -> "NumberGuesser":
        return NumberGuesser()

    def next_guess(self, interval: Interval) -> int:
        return interval.lower + (interval.upper - interval.lower) // 2
